app/presenters/supergroups/news_and_communications.rb

```diff
@@ -26,22 +28,33 @@ def tagged_content
 
     def promoted_content
       items = @content.shift(promoted_content_count)
-      content = format_document_data(items, data_category: "ImageCardClicked")
+      format_document_data(items, data_category: "ImageCardClicked")
+    end
+
+  private
 
-      content.each do |document|
-        document_image = news_item_photo(document[:link][:path])
-        document[:image] = {
-          url: document_image["url"],
-          alt: document_image["alt_text"],
-          context: document_image["context"]
+    def format_document_data(documents, data_category: nil)
+      documents.each.with_index(1).map do |document, index|
```

Comment:
I had to double check the Ruby API documentation to confirm that `1` is an offset. Could we extract this to provide a little more context?

Reply:
👍